### PR TITLE
binaryen.js: Remove closure flags that set the ES version

### PR DESCRIPTION
**Pull request by kripken**
Emscripten sets that itself these days.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -468,7 +468,6 @@ if(EMSCRIPTEN)
   target_link_libraries(binaryen_wasm "-msign-ext")
```
Comment (sbc100) [inline CMakeLists.txt:468]: I think sign-ext is the default these days so you can probably remove this line too

**Comment by kripken** [comment]
Hmm, this still errors, so that wasn't it. Any other theories @sbc100 ?

**Comment by sbc100** [comment]
I see the correct flags in there `--language_in ECMASCRIPT_2021 --language_out NO_TRANSPILE` .. but then someone is injecting these incorrect flags later `language_in=ECMASCRIPT6 --language_out=ECMASCRIPT6 `

**Comment by kripken** [comment]
I bisected the regression to "Leverage coalescing operators. NFC https://github.com/emscripten-core/emscripten/pull/20531"

**Comment by kripken** [comment]
I also may have found the problem here, those flags were in a second place.

**Pull request merged**